### PR TITLE
http-client: add option to refresh authentication on 401/403 errors

### PR DESCRIPTION
**Pull request by RonaldEAM**

### CODE REVIEW

File: packages/integration-sdk-http-client/src/types.ts [hunk 1]
```diff
@@ -18,7 +18,10 @@ export interface RetryOptions {
   factor: number;
   timeout: number;
   timeoutMaxAttempts: number;
-  handleError: (
+  /**
+   * @deprecated Override `retryErrorHandler` protected method instead.
```
Comment (VDubber) [inline packages/integration-sdk-http-client/src/types.ts:22]: 🔥

File: packages/integration-sdk-http-client/src/client.ts [hunk 2]
```diff
@@ -335,7 +312,11 @@ export abstract class BaseAPIClient {
           timeout: this.retryOptions.timeout,
           factor: this.retryOptions.factor,
           handleError: async (err, context) => {
-            await this.retryOptions.handleError(err, context, this.logger);
+            if (this.retryOptions.handleError) {
```
[the remark this answers is not in the document]
Comment (VDubber) [reply packages/integration-sdk-http-client/src/client.ts:315]: You are already one setup ahead of me!

File: packages/integration-sdk-http-client/src/client.ts [hunk 3]
```diff
@@ -322,9 +297,11 @@ export abstract class BaseAPIClient {
             } else {
               error = await fatalRequestError(requestErrorParams);
             }
-            for await (const _chunk of response.body) {
-              // force consumption of body to avoid memory leaks
-              // https://github.com/node-fetch/node-fetch/issues/83
+            if (response.body) {
+              for await (const _chunk of response.body) {
+                // force consumption of body to avoid memory leaks
+                // https://github.com/node-fetch/node-fetch/issues/83
```
Comment (VDubber) [inline packages/integration-sdk-http-client/src/client.ts:303]: I can't wait until we are on node 21 and we can start using Global Fetch.

**Review by VDubber — APPROVED** [review]

File: packages/integration-sdk-http-client/src/client.ts (hunk 3 above)
Comment (RonaldEAM) [reply packages/integration-sdk-http-client/src/client.ts:303]: Yes! Me too

**Review by VDubber — APPROVED** [review]

**Pull request merged**